File: DSA/AnshulJain_590014934/Day_82_Question_3.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None
def insert(root, val):
    if not root:
        return Node(val)
    if val < root.val:
        root.left = insert(root.left, val)
    else:
        root.right = insert(root.right, val)
    return root
def transform(root, acc=0):
    if not root:
        return acc
    acc = transform(root.right, acc)
    root.val += acc
    acc = root.val
    return transform(root.left, acc)
def inorder(root):
    if not root:
        return []
    return inorder(root.left) + [root.val] + inorder(root.right)
arr = list(map(int, input().split()))
tree = None
for x in arr:
    tree = insert(tree, x)
transform(tree)
print(*inorder(tree))

File: DSA/RATNA_KHANNA_590013141/DAY_82_QUESTION_3.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* left;
    struct Node* right;
};

struct Node* newNode(int value) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = value;
    node->left = NULL;
    node->right = NULL;
    return node;
}

struct Node* insert(struct Node* root, int value) {
    if (root == NULL) return newNode(value);
    if (value < root->data) root->left = insert(root->left, value);
    else root->right = insert(root->right, value);
    return root;
}

void convertToGreaterTree(struct Node* root, int* sum) {
    if (root == NULL) return;
    convertToGreaterTree(root->right, sum);
    *sum += root->data;
    root->data = *sum;
    convertToGreaterTree(root->left, sum);
}

void inorder(struct Node* root) {
    if (root == NULL) return;
    inorder(root->left);
    printf("%d ", root->data);
    inorder(root->right);
}

int main() {
    int n, i, value, sum = 0;
    struct Node* root = NULL;
    printf("Enter number of nodes: ");
    scanf("%d", &n);
    printf("Enter node values: ");
    for (i = 0; i < n; i++) {
        scanf("%d", &value);
        root = insert(root, value);
    }
    convertToGreaterTree(root, &sum);
    printf("Inorder Traversal of Greater Tree: ");
    inorder(root);
    printf("\n");
    return 0;
}
